File: CppSolution/JumpGame.cpp
#include <iostream>
using namespace std;

class Solution 
{
public:
    bool canJump(int A[], int n) 
    {
    	if (n == 0)
    		return false;
    	if (n == 1)
    		return true;
    	int reach = 0;
    	for (int i = 0; i < n && i <= reach; i++)
    	{
    		reach = max(A[i] + i, reach);
    	}
    	if (reach >= n - 1)
    		return true;
    	return false;
    }
};

int main()
{
	return 0;
}

File: JavaSolution/src/Microsoft/JumpGame.java
package Microsoft;

/**
 * Created by xyunpeng on 2/26/16.
 */
public class JumpGame {
    public boolean canJump(int[] nums) {
        if (nums == null || nums.length < 2) {
            return true;
        }

        int maxLen = 0;
        for (int i = 0; i < nums.length && i <= maxLen; i++) {
            maxLen = Math.max(maxLen, nums[i] + i);
        }

        return maxLen >= nums.length - 1;
    }
}
